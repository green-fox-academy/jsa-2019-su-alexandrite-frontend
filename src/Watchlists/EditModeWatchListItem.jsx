import PropTypes from 'prop-types';
import React from 'react';
import {
  Text, View, TouchableHighlight,
} from 'react-native';
import { MaterialIcons } from '@expo/vector-icons';
import styles from './styles';

export default function WatchListItem({
  isChecked,
  onSelect,
  ticker,
}) {
  return (
    <TouchableHighlight
      underlayColor="rgba(0,0,0,0.1)"
      onPress={onSelect}
    >
      <View style={styles.watchListItem}>
        <MaterialIcons
          style={[styles.checkButton, { color: isChecked ? '#666' : '#eee' }]}
          name={isChecked ? 'check-box' : 'check-box-outline-blank'}
        />
        <View style={styles.watchListItemColumn}>
          <Text style={[styles.watchListItemColumnText, { fontWeight: 'bold', color: '#566ed3' }]}>
            {ticker}
          </Text>
        </View>
      </View>
    </TouchableHighlight>
  );
}

WatchListItem.propTypes = {
  ticker: PropTypes.string.isRequired,
  isChecked: PropTypes.bool.isRequired,
  onSelect: PropTypes.func.isRequired,
};
