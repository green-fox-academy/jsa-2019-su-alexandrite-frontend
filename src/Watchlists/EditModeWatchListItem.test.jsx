import React from 'react';
import renderer from 'react-test-renderer';

import EditModeWatchListItem from './EditModeWatchListItem';

describe('<EditModeWatchListItem />', () => {
  it('Should render properly when checkbox is not checked', () => {
    const tree = renderer.create(
      <EditModeWatchListItem
        isChecked={false}
        onSelect={jest.fn()}
        ticker="APPL"
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render properly when checkbox is checked', () => {
    const tree = renderer.create(
      <EditModeWatchListItem
        isChecked
        onSelect={jest.fn()}
        ticker="APPL"
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
