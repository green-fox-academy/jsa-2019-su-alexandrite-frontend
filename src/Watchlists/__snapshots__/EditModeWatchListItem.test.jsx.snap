// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditModeWatchListItem /> Should render properly when checkbox is checked 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "#eee",
        "borderBottomWidth": 1,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 10,
      }
    }
  >
    <Text />
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#666",
              "fontSize": 12,
            },
            Object {
              "color": "#566ed3",
              "fontWeight": "bold",
            },
          ]
        }
      >
        APPL
      </Text>
    </View>
  </View>
</View>
`;

exports[`<EditModeWatchListItem /> Should render properly when checkbox is not checked 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "#eee",
        "borderBottomWidth": 1,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 10,
      }
    }
  >
    <Text />
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#666",
              "fontSize": 12,
            },
            Object {
              "color": "#566ed3",
              "fontWeight": "bold",
            },
          ]
        }
      >
        APPL
      </Text>
    </View>
  </View>
</View>
`;
